### ISSUE TITLE
Date parser removes substrings from day of week

### ISSUE THREAD

**Issue by JorisKimpe** [bkiers/Liqp#313]
Because of the newly added array toBeRemoved in liqp.filters.date.Parser (version 0.9.1.2), the parsing of dates fails.
`private static final String[] toBeRemoved = new String[] { "st", "nd", "rd", "th" };`

When we eg have a date like "Monday 25 November 2024 09:58:20", the normalized string to parse becomes "moay 25 november 2024 09:58:20".  The string to remove should only be applied on days like 1st, 2nd, 3d or 4th!  Not on the complete string to parse:
   
```
public static ZonedDateTime parse(String str, Locale locale, ZoneId defaultZone) {
        String normalized = str.toLowerCase();

        for(String value : toBeRemoved) {
            normalized = normalized.replace(value, "");
        }
```

**Comment by msangel** [comment]
True. I have simple fix for this:
```java
    private static final Map<String, String> toBeReplaced = new HashMap<String, String>() {{
            this.put("1st", "1");
            this.put("2nd", "2");
            this.put("3rd", "3");
            this.put("4th", "4");
            this.put("5th", "5");
            this.put("6th", "6");
            this.put("7th", "7");
            this.put("8th", "8");
            this.put("9th", "9");
            this.put("0th", "0");
    }};

``` 

and later:
```
        for(Map.Entry<String, String> kv : toBeReplaced.entrySet()) {
            normalized = normalized.replace(kv.getKey(), kv.getValue());
        }
```

also started my work on new date parser (from scratch as previous code I had there has been lost due to computer change)

**Comment by msangel** [comment]
Fast fix should be as `liqp-0.9.1.3` version tomorrow
